Library/Homebrew/cmd/list.rb

```diff
@@ -40,6 +40,12 @@ def list_args
       switch "--pinned",
              description: "List only pinned formulae, or only the specified (pinned) "\
                           "formulae if <formula> are provided. See also `pin`, `unpin`."
+      switch "--requested",
+             depends_on:  "--formula",
+             description: "List only formulae that were installed on request."
+      switch "--unrequested",
+             depends_on:  "--formula",
+             description: "List formulae that were installed only to satisfy dependencies."
```

Comment:
```suggestion
      switch "-r", "--installed-on-request",
             depends_on:  "--formula",
             description: "List only formulae that were installed on request."
      switch "-p", "--installed-as-dependency",
             depends_on:  "--formula",
             description: "List formulae that were installed only to satisfy dependencies."       
```

To match `brew leaves`. Alternatively, rename and deprecate the long versions on `brew leaves`.

I do wonder, though, whether it's useful to have these arguments on both. Will they not return the same thing?

Reply:
`brew leaves` only lists formulae that don't have dependents, while `brew list` lists all formulae. If you wanted, `brew leaves` could be a switch for `brew list`.

Reply:
A non-leaf formula can still be installed-on-request. `python@3.9` is a typical example.